### PR TITLE
Add Name to PodOptions

### PR DESCRIPTION
**Pull request by kale-amruta**
## Change Overview

Add an option to have fixed name for a pod

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by akankshakumari393** [comment]
LGTM

File: pkg/kube/pod.go [hunk 1]
```diff
@@ -129,6 +130,11 @@ func CreatePod(ctx context.Context, cli kubernetes.Interface, opts *PodOptions)
 		Spec: patchedSpecs,
 	}
 
+	// Add fixed name to a pod
+	if opts.Name != "" {
```
Comment (ankitjain235) [inline pkg/kube/pod.go:134]: Do we need to not set `GenerateName` if `Name` is set? Or would the `Name` be chosen as precedence always? Should we fail pod creation if both are set?

**Review by ankitjain235 — COMMENTED** [review]
Minor question, looks good otherwise.

File: pkg/kube/pod.go (hunk 1 above)
Comment (kale-amruta) [reply pkg/kube/pod.go:134]: kubernetes give Name as precedence over GenerateName always. Hence it doesnt matter even if the field is set

File: pkg/kube/pod.go [hunk 2]
```diff
@@ -129,6 +130,11 @@ func CreatePod(ctx context.Context, cli kubernetes.Interface, opts *PodOptions)
 		Spec: patchedSpecs,
 	}
 
+	// Add fixed name to a pod
```
Comment (pavannd1) [inline pkg/kube/pod.go:133]:
```suggestion
	// Override `GenerateName` if `Name` option is provided
```

**Review by pavannd1 — APPROVED** [review]
Minor comment update. LG otherwise

**Pull request merged**